### PR TITLE
[SPARK-46363][CORE][SQL] Improve the java text block with java15 feature.

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Now, Spark upgraded to Java17 which contains many new feature. The text block is a feature released in Java15.
When using text blocks, the code looks more concise.

For example,
```
String timeError = "Time must be specified as seconds (s), " +
   "milliseconds (ms), microseconds (us), minutes (m or min), hour (h), or day (d). " +
   "E.g. 50s, 100ms, or 250us.";
```
Can be changed to.
```
String timeError = """
   Time must be specified as seconds (s), milliseconds (ms), microseconds (us), \
   minutes (m or min), hour (h), or day (d). E.g. 50s, 100ms, or 250us.""";
```

This PR is a initial one used for discussion and decision-making.
Shall we improve and simplify the long string with  java15 text block?

### Why are the changes needed?
Using [JEP 378: Text Blocks](https://openjdk.org/jeps/378) can bring the following benefits:

- Simplify the task of writing Java programs by making it easy to express strings that span several lines of source code, while avoiding escape sequences in common cases.
- Enhance the readability of strings in Java programs that denote code written in non-Java languages.
- Support migration from string literals by stipulating that any new construct can express the same set of strings as a string literal, interpret the same escape sequences, and be manipulated in the same ways as a string literal.
- Add escape sequences for managing explicit white space and newline control.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
GA.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
@dongjoon-hyun @srowen @LuciferYang What are your suggestions?

**Comment by beliefer** [comment]
cc @MaxGekk

**Review by MaxGekk — COMMENTED** [review]
SGTM since we don't need to preserve backward compatibility to < java 15.

**Comment by LuciferYang** [comment]
fine to me, I think we should push it forward. @beliefer

File: common/network-common/src/main/java/org/apache/spark/network/TransportContext.java [hunk 1]
```diff
@@ -272,8 +272,9 @@ private SSLFactory createSslFactory() {
             conf.sslRpctrustStoreReloadIntervalMs())
           .build();
       } else {
-        logger.error("RPC SSL encryption enabled but keys not found!" +
-          "Please ensure the configured keys are present.");
+        logger.error("""
+                     RPC SSL encryption enabled but keys not found!\
```
Comment (LuciferYang) [inline common/network-common/src/main/java/org/apache/spark/network/TransportContext.java:276]: Indentation?

File: common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java [hunk 2]
```diff
@@ -162,10 +162,11 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         if (e.state() == IdleState.ALL_IDLE && isActuallyOverdue) {
           if (responseHandler.hasOutstandingRequests()) {
             String address = getRemoteAddress(ctx.channel());
-            logger.error("Connection to {} has been quiet for {} ms while there are outstanding " +
-              "requests. Assuming connection is dead; please adjust" +
-              " spark.{}.io.connectionTimeout if this is wrong.",
-              address, requestTimeoutNs / 1000 / 1000, transportContext.getConf().getModuleName());
+            logger.error("""
+              Connection to {} has been quiet for {} ms while there are outstanding requests. \
```
Comment (LuciferYang) [inline common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java:166]: `{}` will not be replaced by parameters in the `formatted`, right?

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java [hunk 3]
```diff
@@ -228,9 +228,10 @@ AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
     AppShuffleMergePartitionsInfo shufflePartitionsWithMergeId =
       shuffles.compute(shuffleId, (id, mergePartitionsInfo) -> {
         if (mergePartitionsInfo == null) {
-          logger.info("{} attempt {} shuffle {} shuffleMerge {}: creating a new shuffle " +
-              "merge metadata", appShuffleInfo.appId, appShuffleInfo.attemptId, shuffleId,
-              shuffleMergeId);
+          logger.info("""
```
Comment (LuciferYang) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:231]: ditto

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java [hunk 4]
```diff
@@ -247,9 +248,11 @@ AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
             AppAttemptShuffleMergeId currrentAppAttemptShuffleMergeId =
                 new AppAttemptShuffleMergeId(appShuffleInfo.appId, appShuffleInfo.attemptId,
                     shuffleId, latestShuffleMergeId);
-            logger.info("{}: creating a new shuffle merge metadata since received " +
-                "shuffleMergeId {} is higher than latest shuffleMergeId {}",
-                currrentAppAttemptShuffleMergeId, shuffleMergeId, latestShuffleMergeId);
+            logger.info("""
```
Comment (LuciferYang) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:251]: ditto

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java [hunk 5]
```diff
@@ -281,14 +284,14 @@ AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
...
       } catch (IOException e) {
-        logger.error("{} attempt {} shuffle {} shuffleMerge {}: cannot create merged shuffle " +
-            "partition with data file {}, index file {}, and meta file {}", appShuffleInfo.appId,
-            appShuffleInfo.attemptId, shuffleId, shuffleMergeId, dataFile.getAbsolutePath(),
-            indexFile.getAbsolutePath(), metaFile.getAbsolutePath());
-        throw new RuntimeException(
-          String.format("Cannot initialize merged shuffle partition for appId %s shuffleId %s "
-            + "shuffleMergeId %s reduceId %s", appShuffleInfo.appId, shuffleId, shuffleMergeId,
-              reduceId), e);
+        logger.error("""
```
Comment (LuciferYang) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:287]: ditto

File: common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java (hunk 2 above)
Comment (beliefer) [reply common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java:166]: Yeah. It's the format for logger. Let's revert it.

**Comment by beliefer** [comment]
Based on the suggestion from @cloud-fan, I close this PR. I will reopen this PR if need.

**Pull request closed without merging**